packages/docusaurus-plugin-content-docs/src/docs.ts

```diff
@@ -47,28 +48,35 @@ type LastUpdateOptions = Pick<
   'showLastUpdateAuthor' | 'showLastUpdateTime'
 >;
 
-async function readLastUpdateData(
+function readLastUpdateData(
   filePath: string,
   options: LastUpdateOptions,
-): Promise<LastUpdateData> {
+  lastUpdateFrontMatter: FileChange | undefined,
+): LastUpdateData {
   const {showLastUpdateAuthor, showLastUpdateTime} = options;
   if (showLastUpdateAuthor || showLastUpdateTime) {
     // Use fake data in dev for faster development.
     const fileLastUpdateData =
       process.env.NODE_ENV === 'production'
-        ? await getFileLastUpdate(filePath)
+        ? getFileLastUpdate(filePath)
```

Comment:
keep await

If both lastUpdate and author are provided through frontmatter, we probably want to return early and avoid calling `getFileLastUpdate` considering the values will never be used